### PR TITLE
feat(cli): add delete-users-by-role command

### PR DESCRIPTION
**Pull request by kariae**
# Add delete-users-by-role command

This PR introduces a new WP-CLI command `delete-users-by-role` that allows for bulk deletion of WordPress users based on their roles.

## Features
- Delete users by specifying roles to delete:
  ```bash
  wp newspack-migration-tools delete-users-by-role --roles=subscriber,author
  ```
- Delete users by excluding specific roles:
  ```bash
  wp newspack-migration-tools delete-users-by-role --exclude-roles=administrator,editor
  ```
- Optional post reassignment:
  ```bash
  wp newspack-migration-tools delete-users-by-role --roles=subscriber --reassign-to=1
  ```
- Dry-run mode for safety:
  ```bash
  wp newspack-migration-tools delete-users-by-role --roles=subscriber --dry-run
  ```

## Implementation Details
- Validates roles before processing
- Handles post reassignment/deletion
- Includes detailed logging
- Includes comprehensive test suite
- Includes workaround for [Co-Authors Plus issue #1096](https://github.com/Automattic/Co-Authors-Plus/issues/1096)

## Testing Instructions
1. Install and activate the plugin
2. Run the command with `--dry-run` to preview changes:
   ```bash
   wp newspack-migration-tools delete-users-by-role --roles=subscriber --dry-run
   ```
3. Run without `--dry-run` to perform actual deletion
4. Test with both `--roles` and `--exclude-roles` parameters
5. Test with and without `--reassign-to` parameter

## QA Checklist
- [x] Code follows WordPress coding standards
- [x] All tests pass
- [x] Documentation is clear and complete
- [x] Command handles edge cases gracefully

### CODE REVIEW

File: tests/Command/TestUsersMigrator.php [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+
+		// Assert that no users were deleted
+		$authors = get_users( [ 'role' => 'author' ] );
+		$this->assertEquals( 1, count( $authors ) );
+	}
+
+	/**
+	 * Clean up after all tests.
+	 */
+	public function tearDown(): void {
```
Comment (naxoc) [inline tests/Command/TestUsersMigrator.php]: Does all this (maybe except for the wp hook/removal) not happen automatically? I've never had to tear down stuff like this, but I could be wrong.

**Review by jorostoyanov — APPROVED** [review]
Awesome! Works exactly as described 🙌

**Review by jorostoyanov — CHANGES_REQUESTED** [review]
Hey @kariae  👋 

Do you think it's worth adding a confirm step if the `--reassign-to` option is not passed? Otherwise, we can accidentally delete Posts this way.

I just had this happened to me and wanted to make sure we prevent this from happening again 🙌

**Comment by kariae** [comment]
Hey @jorostoyanov 👋 

[531c355](https://github.com/Automattic/newspack-migration-tools/pull/57/commits/531c35592c3aef6f2913f746f47164d2890e0f7a) forces setting the `reassign-to` parameter so we can avoid accidentally deleting posts.

File: tests/Command/TestUsersMigrator.php (hunk 1 above)
Comment (kariae) [reply tests/Command/TestUsersMigrator.php]: Good point. I deleted the tearDown method.

**Comment by kariae** [comment]
@jorostoyanov @naxoc can I get a final check on this, please?

**Review by naxoc — APPROVED** [review]
This looks good! A small thing: no need to cleanup users in the test I think. Each function is run on a clean DB.
